packages/sdk/src/cross-chain-messenger.ts

```diff
@@ -805,32 +803,27 @@ export class CrossChainMessenger {
           const withdrawalHash = hashLowLevelMessage(withdrawal)
 
           // Grab the proven withdrawal data.
-          const provenWithdrawal =
-            await this.contracts.l1.OptimismPortal2.provenWithdrawals(
-              withdrawalHash
-            )
+          const provenWithdrawal = (await this.getProvenWithdrawal(
+            withdrawalHash
+          )) as FPACProvenWithdrawal
 
-          // Attach to the FaultDisputeGame.
-          const game = new ethers.Contract(
-            provenWithdrawal.disputeGameProxy,
-            getContractInterfaceBedrock('FaultDisputeGame'),
-            this.l1SignerOrProvider
-          )
+          // Sanity, check should've already happened above but do it just in case.
+          if (provenWithdrawal === null) {
+            return MessageStatus.READY_TO_PROVE
```

Comment:
```suggestion
          if (provenWithdrawal === null) {
            console.warn('Unexpected code path reached in getMessageStatus. Returning `READY_TO_PROVE`')
            return MessageStatus.READY_TO_PROVE
```
If it should never happen we should throw or console.warn here. I would tend to throw here but not super opinionated on which one.

Reply:
happy to keep the comment too

Reply:
Actually I guess some wierd reorg type race condition could happen in which case console.warn is better